### PR TITLE
S04 us01favorite recipe

### PR DESCRIPTION
**Pull request by plbonnier**

### CODE REVIEW

**Review by Aude711 — APPROVED** [review]

**Review by nickyrasa — APPROVED** [review]

File: src/Entity/User.php [hunk 1]
```diff
@@ -200,4 +205,28 @@ public function removeRecipe(Recipe $recipe): static
 
         return $this;
     }
+
+    /**
+     * @return Collection<int, Recipe>
+     */
+    public function getFavoritelist(): Collection
```
Comment (StoicFrenchDev) [inline src/Entity/User.php:212]: You're going to hate me.
**cameCase!**
Of course make sure to update the rest of the file and the ones who are impacted by the changes.

File: src/Repository/UserRepository.php [hunk 2]
```diff
@@ -38,29 +39,16 @@ public function upgradePassword(PasswordAuthenticatedUserInterface $user, string
         $this->getEntityManager()->persist($user);
         $this->getEntityManager()->flush();
     }
+    public function countLikersByRecipe(Recipe $recipe): int
```
Comment (StoicFrenchDev) [inline src/Repository/UserRepository.php:42]: Wouldn't it make more sense to have this query builder in IdeaRepository?

File: src/Controller/RecipeController.php [hunk 3]
```diff
@@ -103,10 +104,12 @@ public function new(Request $request, MailerInterface $mailer, EntityManagerInte
     }
 
     #[Route('/{id}', name: 'app_recipe_show', methods: ['GET'])]
-    public function show(Recipe $recipe): Response
+    public function show(Recipe $recipe, UserRepository $userRepository): Response
     {
+        $totalLikers = $userRepository->countLikersByRecipe($recipe);
         return $this->render('recipe/show.html.twig', [
             'recipe' => $recipe,
+            'totalLikers' => $totalLikers
```
Comment (StoicFrenchDev) [inline src/Controller/RecipeController.php:112]: Not sending the liked recipes? Isn't it part of the US?

**Review by StoicFrenchDev — CHANGES_REQUESTED** [review]
It's monday, feeling picky.

File: src/Repository/UserRepository.php (hunk 2 above)
Comment (plbonnier) [reply src/Repository/UserRepository.php:42]: I don't understand why you talk about IdeaRepository and why this one have sense if it's in IdeaRepository?

Comment (StoicFrenchDev) [reply src/Repository/UserRepository.php:42]: RecipeRepository*

File: templates/components/Favoritelist.html.twig [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<div{{attributes}}>
+	<button class="btn p-0" data-action="live#action" data-action-name="toggle">
+		{% if app.user %}
+			{% if app.user.favoritelist.contains(this.recipe) %}
```
Comment (StoicFrenchDev) [inline templates/components/Favoritelist.html.twig:4]: Missed this "l"

**Review by StoicFrenchDev — CHANGES_REQUESTED** [review]

**Review by StoicFrenchDev — APPROVED** [review]

**Pull request merged**